### ISSUE TITLE
Make users read CoC before sending username to server

### ISSUE THREAD

**Issue by Arch-Shaman** [ZeroK-RTS/Chobby#771]

**Comment by sprunk** [comment]
Everybody just scrolls down and clicks "ok" without reading anyway though?

**Comment by Licho1** [comment]
Yeah .. if you want people to read something it would have to be something super short like this:

 - [ ] I will play it nice and won't annoy other people by harrassing or teamkilling.
 - [ ] I won't use nazi, homophobe, or otherwise offensive language around here. 
 - [ ] I will always listen to admins, they rule here. If they say so they can ban me forever.

**Comment by Arch-Shaman** [comment]
By hitting the okay button I agree to the following:
 * My username is not overtly offensive, and is Safe For Work.
 * I won't teamkill, be disruptive, or harass other users.
 * I will not post or send any racist, homophobic, pornographic or other bigoted messages.
 * I will not verbally abuse other members of this community.


Note that you can find a full copy of the code of conduct at http://zero-k.info/mediawiki/index.php?title=Zero-K:Code_of_Conduct

**Comment by Licho1** [comment]
Btw safe for work appears to be some specialized slang phrase.. never heard it or anything like that. Might be local culture. What does it mean?
Are you supposed to play ZK at work but at the same time some name might be offensive to boss? 

I alkso think its part of the "i will not post blah blah" stence. I would rather notify them that this is not a typical commercial product and admins can do whatever they like so they better listen to them.

**Comment by maackey** [comment]
Overall I'm not sure how much this would really help, but I like @Licho1 's suggestion of having a checkbox by each (very short) statement. Otherwise people won't read and skip to the end, making it completely pointless.

"By selecting ok / using this software / etc. I agree to..." -- my eyes immediately glaze over. I can't imagine many would click the external link to a detailed description of rules either. And if they do, it's likely they care about rules and won't cause issues in the first place, or are malignant trolls looking for loopholes or clever ways to toe the line -- low utility in either case.

Since everyone's posting their list, here's my take:
- [ ] I will play nice, and won't cheat, teamkill or harass other people.
- [ ] I will not use bigoted, or other intentionally offensive language.
- [ ] I will not send any material that is illegal, pornographic, or obscene.
- [ ] I will abide by moderators' decisions. 

Also, to answer your question, I think NSFW (not safe for work) was originally meant for porn, but I've seen it being used to tag incredibly tame, innocuous things that aren't even sexual in nature. I'm not sure it has a good, clear definition anymore. Besides, for people working from home during this pandemic it doesn't even matter in the slightest.

**Comment by GoogleFrog** [comment]
I don't think you can make users read anything, and such a popup is pretty bad for friction.

**Comment by Licho1** [comment]
I kind of agree with GoogleFrog. Anyone who joins probably knows whats good and bad and what behavior is offensive. People are also trained to just click accept on any agreements/coc/legal text. 
To be useful it should contain some information/rule thats not obvious. 
Also consider the language barrier, many people joining do not speak English. They can understand common gaming words like "multiplayer" and "skirmish" but will struglly to decipher such sentence and possibly be scared out (call parents, consult legal advisor)

**Comment by sprunk** [comment]
I think the current system works fine. People generally already know how to behave, those who don't generally get a lenient warning first which is a great opportunity to point the rules out.
